client/app/scripts/superdesk-packaging/packaging.js

```diff
@@ -331,9 +331,13 @@
             templateUrl: 'scripts/superdesk-packaging/views/sd-package-items-edit.html',
             link: function(scope, elem, attrs, ngModel) {
                 scope.$on('package:addItems', function(event, data) {
-                    var groupIndex = _.findIndex(scope.list, {id: 'main'});
-                    for (var i = 0; i < data.length; i++) {
-                        scope.list[groupIndex].items.unshift(packages.getReferenceFor(data[i]));
+                    var groupIndex = _.findIndex(scope.list, {id: data.group});
+                    if (groupIndex === -1) {
+                        scope.list.push({id: data.group, items: []});
+                        groupIndex = scope.list.length - 1;
+                    }
+                    for (var i = 0; i < data.items.length; i++) {
+                        scope.list[groupIndex].items.unshift(packages.getReferenceFor(data.items[i]));
                     }
```

Comment:
@eozan please, the comment was meant constructively, and was not a blocker for merging in any way, I tried my best to make that clear (please correct me if that was not the case). Using `track by` and one time bindings, on the other hand, _can_ have noticeable effects on the UI performance, and it's important that we pay attention to that (these two things still appear in PRs quite often), esp. because some more heavy parts of the UI sometimes already feel laggy.

26M ops/s vs. 1M ops/s does not make a noticeable difference for typical arrays that don't contain more than e.g. 300 elements (it's less than 0.3 ms), but having too many `$watches` on the page, on the other hand, hinders performance, and that is definitely not nitpicking.

No need for a harsh tone really, does not help with anything, but in case you feel disturbed by anything, I kindly invite you to send an email, so that we can sort it out.


(_"Always assume the best"_ - Micz)